modules/ethereum/src/lib.rs

```diff
@@ -480,10 +499,94 @@ impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
 #[derive(Default)]
 struct BridgeStorage<T>(sp_std::marker::PhantomData<T>);
 
-impl<T> BridgeStorage<T> {
+impl<T: Trait> BridgeStorage<T> {
+	/// Create new BridgeStorage.
 	pub fn new() -> Self {
 		BridgeStorage(sp_std::marker::PhantomData::<T>::default())
 	}
+
+	/// Prune old blocks.
+	fn prune_blocks(&self, mut max_blocks_to_prune: u64, finalized_number: u64, prune_end: Option<u64>) {
+		let pruning_range = BlocksToPrune::get();
+		let mut new_pruning_range = pruning_range.clone();
+
+		// update oldest block we want to keep
+		if let Some(prune_end) = prune_end {
+			new_pruning_range.oldest_block_to_keep = prune_end;
+		}
+
+		// start pruning blocks
+		let begin = new_pruning_range.oldest_unpruned_block;
+		let end = new_pruning_range.oldest_block_to_keep;
+		for number in begin..end {
```

Comment:
This may panic if `begin > end` - we don't seem to check the `prune_end` before applying it here.